infoFileComparison/src/main/scala/za/co/absa/hermes/infoFileComparison/config/TypesafeConfig.scala

```diff
@@ -0,0 +1,15 @@
+package za.co.absa.hermes.infoFileComparison.config
+
+import com.typesafe.config.{Config, ConfigFactory}
+
+class TypesafeConfig(path: Option[String]) extends InfoFileComparisonConfig {
```

Comment:
Just an optional suggestion.
If the constructor parameter is `Config`, then it is up to the user to construct a configuration from a file, from a map or in the default way. And configuration class hierarchy won't be needed. It could be just a case class holding all required configs with a companion object that creates an instance of configuration from a `TypeSafe` `Config`.

Reply:
This is a really nice suggestion. Let me think about it. If I can't find any negatives I will most likely implement.